### PR TITLE
MANAGER-15346 : refactor(*): add retry on universes and availability requests

### PR DESCRIPTION
**Pull request by tibs245**
| Question         | Answer
| ---------------- | ---
| Branch?          | `develop`
| Bug fix?         | no
| New feature?     | no
| Breaking change? | no
| Tickets          | Refactor #MANAGER-15346
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [ ] ~Only FR translations have been updated~
- [ ] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] ~Breaking change is mentioned in relevant commits~

### CODE REVIEW

File: packages/manager/apps/container/src/hooks/useUniverses.ts [hunk 1]
```diff
@@ -29,17 +26,17 @@ export async function fetchUniverses(): Promise<Universe[]> {
 }
 
 export function useUniverses() {
-  const [universes, setUniverses] = useState<Universe[]>([]);
+  const { data: universes, isLoading, isSuccess } = useQuery({
+    queryKey: ["universe"],
```
Comment (darsene) [inline packages/manager/apps/container/src/hooks/useUniverses.ts:30]:
```suggestion
    queryKey: ["universes"],
```

File: packages/manager/apps/container/src/hooks/useUniverses.ts [hunk 2]
```diff
@@ -29,17 +26,17 @@ export async function fetchUniverses(): Promise<Universe[]> {
...
-  }, []);
 
   return {
-    getUniverses: (): Universe[] => universes,
-    isLoading: () => !universes?.length,
+    getUniverses: (): Universe[] => universes ?? [],
+    isLoading: () => isLoading,
     getHubUniverse: () => {
-      return find(universes, { universe: 'hub' });
+      return (universes ?? []).find(({ universe }) => universe === 'hub');
```
Comment (darsene) [inline packages/manager/apps/container/src/hooks/useUniverses.ts:39]:
```suggestion
      return universes?.find(({ universe }) => universe === 'hub');
```

**Review by anooparveti — CHANGES_REQUESTED** [review]
**question:** Do we have any ticket which requires this change?

**Comment by darsene** [comment]
> **question:** Do we have any ticket which requires this change?

@anooparveti it's to fix the issue with BFF ran locally 

```
{
    "message": "CDN's redirect mechanism has been triggered due to many calls to the same API route in a small amount of time. You probably need to optimize your code to avoid this.",
    "class": "Client::TooManyRequests"
}
```

**Review by darsene — APPROVED** [review]

**Comment by anooparveti** [comment]
Can you please create a ticket?

**Comment by tibs245** [comment]
> Can you please create a ticket?

Ticket created 🚀 MANAGER-15346

**Review by darsene — DISMISSED** [review]

**Review by pauldkn — APPROVED** [review]

**Review by anooparveti — APPROVED** [review]

**Review by darsene — APPROVED** [review]

**Review by pauldkn — APPROVED** [review]

**Pull request merged**